File: app/src/main/java/com/binatestation/kickstart/fragments/TabLayoutFragment.java
package com.binatestation.kickstart.fragments;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.design.widget.TabLayout;
import android.support.v4.app.Fragment;
import android.support.v4.view.ViewPager;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.binatestation.kickstart.R;
import com.binatestation.kickstart.adapters.ViewPagerAdapter;

/**
 * Created by RKR on 30-08-2018.
 * ViewPagerFragment.
 */
public abstract class TabLayoutFragment extends BaseFragment {
    private static final String TAG = "ViewPagerFragment";

    private ViewPagerAdapter mViewPagerAdapter;
    private ViewPager mViewPager;
    private TabLayout mTabLayout;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_tab_layout, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        intView(view);
    }

    private void intView(@NonNull View view) {
        mTabLayout = view.findViewById(R.id.tab_layout);
        mViewPager = view.findViewById(R.id.view_pager);
        mTabLayout.setupWithViewPager(mViewPager);
        mViewPager.setAdapter(mViewPagerAdapter = new ViewPagerAdapter(getChildFragmentManager()));
    }

    public void addFragment(Fragment fragment, String title) {
        Log.d(TAG, "addFragment() called with: fragment = [" + fragment + "], title = [" + title + "]");
        if (mViewPagerAdapter != null) {
            mViewPagerAdapter.addFrag(fragment, title);
            mViewPagerAdapter.notifyDataSetChanged();
        }
    }

    public ViewPager getViewPager() {
        return mViewPager;
    }

    public TabLayout getTabLayout() {
        return mTabLayout;
    }
}

File: kickoff/src/main/java/com/binatestation/android/kickoff/utils/fragments/TabLayoutFragment.kt
/*
 * Copyright (c) 2021. Binate Station Private Limited. All rights reserved.
 */

@file:Suppress("MemberVisibilityCanBePrivate")

package com.binatestation.android.kickoff.utils.fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.viewpager.widget.ViewPager
import com.binatestation.android.kickoff.R
import com.binatestation.android.kickoff.utils.adapters.ViewPagerAdapter
import com.google.android.material.tabs.TabLayout

/**
 * Created by RKR on 30-08-2018.
 * ViewPagerFragment.
 */
abstract class TabLayoutFragment : Fragment() {

    private var mViewPagerAdapter: ViewPagerAdapter? = null
    var viewPager: ViewPager? = null
    var tabLayout: TabLayout? = null

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return inflater.inflate(R.layout.fragment_tab_layout, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        viewPager = view.findViewById(R.id.view_pager)
        tabLayout = view.findViewById(R.id.tab_layout)
        intView()
    }

    private fun intView() {
        tabLayout?.setupWithViewPager(viewPager)
        mViewPagerAdapter = ViewPagerAdapter(childFragmentManager)
        viewPager?.adapter = mViewPagerAdapter
    }

    @Suppress("unused")
    fun addFragment(fragment: Fragment, title: String) {
        mViewPagerAdapter?.let { it.addFrag(fragment, title); it.notifyDataSetChanged() }
    }

    /**
     * Function to get count of the fragments
     * @return Int
     */
    fun getCount(): Int {
        return mViewPagerAdapter?.count ?: 0
    }
}
